### PR TITLE
Deploy secure config from Foreman params

### PR DESCRIPTION
**Pull request by GregSutcliffe**
Part1 of deploying the tokens from Foreman. Once this is on disk i'll update the redmine repo to use it.

### CODE REVIEW

File: puppet/modules/redmine/manifests/init.pp [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 #
-class redmine {
+# === Parameters:
+#
+# $secret_token::   Token used in the Rails initializer for session auth, etc
+#
+# $email_password:: Mailgun SMTP access password
+#
+class redmine (
+  String $secret_token   = "token",
```
Comment (ekohl) [inline puppet/modules/redmine/manifests/init.pp]: single quotes

File: puppet/modules/redmine/manifests/init.pp [hunk 2]
```diff
@@ -23,14 +32,26 @@
...
+  file { '/etc/redmine':
+    ensure =>  directory,
+  }
+
+  file { '/etc/redmine/secure_config.yaml':
+    ensure  => present
+    owner   => 'root',
+    group   => 'root'
+    mode    => '0600',
+    content => template('redmine/secure_config.yaml.erb')
```
Comment (ekohl) [inline puppet/modules/redmine/manifests/init.pp]: trailing comma

File: puppet/modules/redmine/manifests/init.pp (hunk 1 above)
Comment (GregSutcliffe) [reply puppet/modules/redmine/manifests/init.pp]: doh

File: puppet/modules/redmine/manifests/init.pp (hunk 2 above)
Comment (GregSutcliffe) [reply puppet/modules/redmine/manifests/init.pp]: Fixed a whole bunch of these

File: puppet/modules/redmine/manifests/init.pp [hunk 3]
```diff
@@ -23,28 +32,40 @@
...
-    home    => $app_root
+    home    => $app_root,
+  }
+
+  file { '/etc/redmine':
+    ensure =>  directory,
+  }
+
+  file { '/etc/redmine/secure_config.yaml':
+    ensure  => present
```
Comment (ekohl) [inline puppet/modules/redmine/manifests/init.pp]: While you're at it: current lint prefers ```file``` over ```present```. Also looks like there's a comma missing.

File: puppet/modules/redmine/manifests/init.pp [hunk 4]
```diff
@@ -23,28 +32,40 @@
...
+    content => template('redmine/secure_config.yaml.erb'),
   }
 
   # TODO: handle cloning the redmine repo and bundle install...
   file { $app_root:
     ensure => directory,
     owner  => 'root',
     group  => 'root',
-    mode   => '0755'
+    mode   => '0755',
```
Comment (ekohl) [inline puppet/modules/redmine/manifests/init.pp]: Mode is duplicated here causing a syntax error.

File: puppet/modules/redmine/manifests/init.pp [hunk 5]
```diff
@@ -23,28 +32,40 @@
...
+  }
+
+  file { '/etc/redmine':
+    ensure =>  directory,
+  }
+
+  file { '/etc/redmine/secure_config.yaml':
+    ensure  => file,
+    owner   => 'root',
+    group   => 'root'
```
Comment (ekohl) [inline puppet/modules/redmine/manifests/init.pp]: Comma is missing causing a syntax error

Comment (GregSutcliffe) [reply puppet/modules/redmine/manifests/init.pp]: Updated.

**Pull request merged**